### ISSUE TITLE
Anime4k/FSR upscaling support.

### ISSUE THREAD

**Issue by CortexReaver** [mkxp-z/mkxp-z#225]
I've got a question, rather then a bug report.
You see all VX Ace (and XP) engine games that I've played has a very small screen resolution, which means when I shave to stretch them on my streams.
I am interested if your "player" supports upscaling like easy RPG maker does:

![2024 101654](https://github.com/user-attachments/assets/99b4972c-6075-44f2-a345-f61f8ecec727)

**Comment by CortexReaver** [comment]
Oh, ok. I've found a "Lanczos3" and "xBRZ" options in the json file. That's really awesome!

Let me reinterpret my question then: will more advanced upscaling algorithms, such as anime4k and FSR, be supported in the future?

**Comment by Splendide-Imaginarius** [comment]
In my experience, you'll generally get better graphical quality by using a non-real-time upscaling algorithm on the game's assets. You can then substitute those into the game via the `enableHires` option in `mkxp.json`. A few people in the mkxp-z community are currently working on game remasters via this approach.

**Comment by CortexReaver** [comment]
Yeah, but this is too much hustle. I am not developer, but a streamer. I've got thousands of VX and XP games on my HDD. Do I need do it to all of them? Sure, I can do this procedure for standard RTP assets, but what about games with unique assets (considering most of them are archived anyway).
I've got some really rare and unfinished projects, that I want to share with others - those are will never get remastered anyway for several reasons: their obscurity, language barrier (got bunch or russian ones) and not to mention their gameplay and asset quality.

For upscaling RPG maker games I am using program called "Losseless scaling" and the results are awesome... the bad thing - only me can see the upscaled version, in obs I have to stretch the picture and tweak it, using existing Scale filters - for the viewer end picture looks like crap anyway.

Also fonts. I thought that these engines use vector fonts, but when I am changing the size of the window - they have jagged edges and artifacts around them. It looks way better with Lossless scaling and anime4k.

**Comment by Splendide-Imaginarius** [comment]
> Yeah, but this is too much hustle.

I hope you realize that it's also a lot of work to add code to mkxp-z for this. Perhaps a lot more work than what I'm suggesting you do.

> I am not developer, but a streamer.

I think we may have different definitions of "developer". You don't need to write any code to use this.

> I've got thousands of VX and XP games on my HDD. Do I need do it to all of them?

I mean, you don't have to use a slow upscaling algorithm for this, you could use one that's fast enough for real-time, and just save to files. How many images do these games have? If you're using a 4x upscaling algorithm that runs at 60fps for 1080p input, then that would be... 3600 images per minute if the input images are 1080p? Is that really too slow to do before you play a given game for the first time?

> Sure, I can do this procedure for standard RTP assets, but what about games with unique assets (considering most of them are archived anyway).

I'm not sure what you mean by "archived". As noted above, if you can do 3600 1080p images per minute, that doesn't seem that bad?

> I've got some really rare and unfinished projects, that I want to share with others - those are will never get remastered anyway for several reasons: their obscurity, language barrier (got bunch or russian ones) and not to mention their gameplay and asset quality.

I play plenty of games with quick-and-dirty upscaled assets (via the approach I suggested), that I don't intend to ever release proper remasters for.

> For upscaling RPG maker games I am using program called "Losseless scaling" and the results are awesome... the bad thing - only me can see the upscaled version, in obs I have to stretch the picture and tweak it, using existing Scale filters - for the viewer end picture looks like crap anyway.

Well, that downside plus the fact that LS appears to be closed-source, so it doesn't really interest me anyway.

> Also fonts. I thought that these engines use vector fonts, but when I am changing the size of the window - they have jagged edges and artifacts around them. It looks way better with Lossless scaling and anime4k.

Fonts are already handled fine by the `enableHires` option. (Did you try it?)

Anyway -- there are some refactors in the works that will hopefully make it easier to add other upscaling algorithms to mkxp-z. Right now it's unpleasant to add more, but if the refactors make it less painful, I might be open to accepting a PR for adding more algorithms.

**Comment by CortexReaver** [comment]
Ok, I've figured a trick to capture an upscaled LS window from old engines. Also I like the final picture more pleasant this way.
Examples:

https://youtu.be/1u6ndCAS1AA
mxkp-z

https://youtu.be/L67lSaP3wFk
LS

**Comment by CortexReaver** [comment]
> I'm not sure what you mean by "archived"

I mean rgssad, rgss2a, rgss3a files. Just to point this exact moment out: not all games have their files just lying in the game folder.

So you need to extract them, edit and put back. Probably you can somehow automatize this process by batch files...
Because you doesn't want to do this for each game by hand: most of them are 5-10 minutes long anyway. So you'll be spending more time tinkering with archives then on actual gameplay (recording footage/video editing).
Not to mention the end size of upscaled assets: the whole folder with these games will double, or even triple, in size.
If not these two things, I might be taking your suggestion into consideration.

TLDR: I rather sacrifice some CPU\GPU resources every time I run game, rather then my time for all of this... and disk space.
Thank you for YOUR time answering me, cheers :)
